google/cloud/spanner/value.h

```diff
@@ -203,12 +203,18 @@ class Value {
     Bytes& operator=(Bytes&&) = default;
     ///@}
 
-    /// @name Equality
+    /// @name Relational operators
     ///@{
     friend bool operator==(Bytes const& a, Bytes const& b) {
       return a.data == b.data;
     }
     friend bool operator!=(Bytes const& a, Bytes const& b) { return !(a == b); }
+    friend bool operator<(Bytes const& a, Bytes const& b) {
```

Comment:
I think less-than on a bytes data type is weird and the semantics are not obvious. Can we provide a link in the docs to where Spanner documents what this means?

Reply:
As discussed offline, it appears there is no clear specification of Spanner type ordering (although it is acknowledged that there should be).  So, just added comments asserting that std::string comparison correctly models Spanner BYTES comparison, and updated the test case to cover the signed -v- unsigned char issue.